include/zephyr/drivers/misc/devmux/devmux.h

```diff
@@ -13,6 +13,7 @@
 #define INCLUDE_ZEPHYR_DRIVERS_MISC_DEVMUX_H_
 
 #include <stdint.h>
+#include <sys/types.h>
```

Comment:
Rather than including a posix header here, in a "below the line" header, let's stick to ISO C.

That should involve a few changes in the following files:
* `include/zephyr/drivers/misc/devmux/devmux.h` (this file)
* `drivers/misc/devmux/devmux.c`
* `tests/drivers/console_switching/src/main.c`

If possible, please split your changes into 2 commits (in this PR) - the first commit for `lib/` and `include/`, and the second commit for `tests/`.